src/pages/PromptPage/PromptPage.jsx

```diff
@@ -0,0 +1,106 @@
+import "./PromptPage.scss";
+import { useState } from "react";
+import Button from "../../components/Button/Button";
+import InputBox from "../../components/InputBox/InputBox";
+import PreviewBox from "../../components/PreviewBox/PreviewBox";
+import { getResponseContent } from "../../utils/openAIcall";
+
+const requestObj = {
+  messages: [
+    {
+      role: "system",
+      content:
+        "You are a friendly assistant, that gives responses to a community organizer appropriate to LinkedIn in JSON format just post content between 200 and 2000 characters in length, no title. The key should be 'content'. Don't include any extra text outside of the post content itself, including hashtags. Don't say you will create the post, just give me the content",
+    },
+    {
+      role: "user",
+      content: "",
+    },
+  ],
+  model: "gpt-3.5-turbo",
+  temperature: 0.02,
+};
+
+const PromptPage = () => {
+  const [inputText, setInputText] = useState("");
+  const [previewText, setPreviewText] = useState("Preview here");
+  const [isEditing, setIsEditing] = useState(false);
+  const [aiResponseLoading, setAiResponseLoading] = useState("");
+
+  const getOpenAIResponse = async () => {
+    setAiResponseLoading(true);
+    setPreviewText("Loading...");
+    requestObj.messages[1].content = inputText;
+    const responseContent = await getResponseContent(requestObj);
+
+    // need to parse JSON to return an object we can work with
+    const parsedContent = JSON.parse(responseContent.content);
```

Comment:
We may have an issue here if the response is an error. When I tested it with an error response, the app crashed. Can we change the order/check if we have response.content before parsing it and setting it to this variable?

Reply:
Great catch, thank you @IguanasEverywhere ! Forgot to test again after adding JSON.parse()